src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs

```diff
@@ -216,26 +217,137 @@ private static unsafe string[] GetCommandLineArgsNative()
             char* lpCmdLine = Interop.Kernel32.GetCommandLine();
             Debug.Assert(lpCmdLine != null);
 
-            int numArgs = 0;
-            char** argvW = Interop.Shell32.CommandLineToArgv(lpCmdLine, &numArgs);
-            if (argvW == null)
-            {
-                ThrowHelper.ThrowOutOfMemoryException();
-            }
+            return SegmentCommandLine(lpCmdLine);
+        }
 
-            try
+        private static unsafe string[] SegmentCommandLine(char* cmdLine)
+        {
+            //---------------------------------------------------------------------
+            // Splits a command line into argc/argv lists, using the VC7 parsing rules.
+            //
+            // This functions interface mimics the CommandLineToArgvW api.
+            //
+            //---------------------------------------------------------------------
+            // NOTE: Implementation-wise, once every few years it would be a good idea to
+            // compare this code with the C runtime library's parse_cmdline method,
+            // which is in vctools\crt\crtw32\startup\stdargv.c.  (Note we don't
+            // support wild cards, and we use Unicode characters exclusively.)
+            // We are up to date as of ~6/2005.
+            //---------------------------------------------------------------------
+
+            // The C# version is ported from C++ version in coreclr
+            // The behavior mimics what MSVCRT does before main,
+            // which is slightly different with CommandLineToArgvW
```

Comment:
```suggestion
            // Parse command line arguments using the rules documented at 
            // https://learn.microsoft.com/cpp/cpp/main-function-command-line-args#parsing-c-command-line-arguments
            //
            // CommandLineToArgvW API cannot be used here since 
            // it has slightly different behavior.
```
Simplify this comment